src/core_plugins/elasticsearch/lib/health_check.js

```diff
@@ -63,6 +63,10 @@ export default function (plugin, server) {
       if (health !== READY) {
         return Promise.delay(REQUEST_DELAY).then(waitUntilReady);
       }
+
+      return new Promise((resolve) => {
+        plugin.status.once('green', resolve);
```

Comment:
Was this a bug?

Reply:
No, but not having it caused a race condition where code that was waiting for the Kibana index to exist would try to access the config document before the `migrateConfig` step was complete, and that step can take a bit now that it uses the savedObjectsClient. This addition just makes sure that `waitUntilReady()` only resolves elasticsearch is really ready.